### ISSUE TITLE
Hashcat not finding the password with mask attack. 

### ISSUE THREAD

**Issue by Rdgeno** [hashcat/hashcat#2226]
I was trying to crack a WPA using mask attack the password is 12 charcters and it was easier to use mask than to create a word list since I know the first four charcters and the rest is just numbers. No matter what I tried it wouldn't get it so I tried a password I already cracked I can crack it with a straight word list, combinatior and hybrid but it wont crack with mask. I tried 2W43?d?d?d?d?d?d?d?d with -1d with and without Marcov I tried increment 8 to 15 all combinations there of and it just wont get it. I even gave it 11 out of 12 charcters nothing. If I give it the password it will get it but that's the only way mask will. I'm using Hashcat on Windows 10 my Gpu wont work with it in Kali but in Windows it works fine. I have been cracking hashes without an issue until I tried mask. I'm using Version 5.1.0 and like I said it works every way except for mask. I'm using the Hashcat gui and trying from command line it just wont get it. I even re-downloaded it and rebuilt it and tested it every way I could it doesnt work. There are no errors it runs exactly like it should it just doesn't get it. I would copy and paste everything here but theres really nothing to see, if anyone wants to see it I can post it. I dont think my system matters either but just in case it's a Hp Compaq Pro 6300 With 24 GB Vengance Ram a 500 GB Seagate Barracuda a 2TB Seagate Firecudda and a 1TB whatever was installed something Japanese I dont remember but that one has nothing to do with Windows that strictly Kali. The C drive has Windows 10 just updated and Hashcat. I also have an Intel i7 3370 other than that there isnt anything to tell, if you've got any ideas I would love to hear them. I thank you for any help you can lend and greatly appreciate it. I would just run the hashes I have with a straight wordlist but I have 9 that I have the first 3 char and know the last are numbers and I have one that I know the first four but have no idea what the last five could be so the mask would be a great help. I ran a straight wordlist for one and that was 14 GB I really dont want to run a seperate wordlist for all of them and hybrid is really really slow it drops down to 300 hashes a second or something like that is a horrible drop, it goes from four hours any other way to weeks with the hybrid. Any way any help would be greatly appreciated.

**Comment by roycewilliams** [comment]
This ... question? ... is better suited for the forums.

**Comment by Rdgeno** [comment]
No the forums would be I'm messing something up or have no idea how to use it. I know how to use it and I'm pretty sure if it's broken that you report it here. It stands to reason all things being equal if 1 mode isnt working and the rest are it's a program issue you know broken. I thought that's what this place was for if I'm wrong I apologize.

**Comment by roycewilliams** [comment]
If you believe that you have identified a reproducible bug, please provide a minimal and precise list of steps necessary to reproduce the issue (without including any sensitive information).

**Comment by Chick3nman** [comment]
This is almost surely an issue with your runtime as I can not reproduce any issues with WPA2 and mask attacks currently. You failed to state what device you are running hashcat on in your original post aside from:

"I'm using Hashcat on Windows 10 my Gpu wont work with it in Kali but in Windows it works fine."

What GPU? What runtime/driver version? Is hashcat giving you warnings/errors when you start it up? What does your complete command look like?

**Comment by Rdgeno** [comment]
Sorry I forgot to list the GPU I was posting on the fly. The GPU is a AMD RX 560 I tried the driver Hashcat forums recommends but it wouldn't work. The driver I'm using does work it's AMD Radeon 19.11.2. For my CPU I'm using the most current opencl from Intel J thought updating would help it didnt. I'm getting the problem on the gui and command line.

**Comment by Rdgeno** [comment]
Sorry I hit the button I'll give you the gui it's easier I click Brute-force I have ?d in charset 1 the other 3 are blank I have set the incriminate or not either way doesnt matter. The hccapx is in the input file space out put is the crack file to write to hash type WPA/WPA2 (2500) no salt no separator. The only box I have checked is Remove found hashes. Workload is on Nightmare abort 90 Markov on or off doesnt matterI dont have any custom commands entered. In the last bit that I have been trying I disabled the pot file because I only have so many cracked hashes and I'm not thinking the hashes are the problem. It doesnt seem to matter either way with that either. I have found another issue that might help if I use the crunch generated wordlist straight some times it gets it some times it doesnt. If I run top204thousand-WPA it wont get it, I have put the passwords into several diffrent lists some of them get it sometimes and some dont get it at all except Rockyou I put the password in there and it nails it every time. Combination nails it every time. So I tried it on my laptop with Windows same issues but it's a pos so no sense talking about that. As far as this system almost every part is new I just put the CPU in today I even tried swapping back to the i5 same problem. Oh on combinator what I did was created a list with just 2W43351 on the left side and a crunch list I generated with the last five numbers starting at 00000 and ending at 99999 nails it every time in seconds, so I made a crunch with 2W and the right file another crunch file it got it again. One thing I forgot the mask on brute is 2W43?d?d?d?d?d?d?d?d I add or subtract the ?d to fit the incriminate. Other than that Windows 10 home v 10.0.18362 build 18362 64 bit SMBIOS v 2.7 UEFI I hope I didnt miss anything and thank you for answering back I sincerely appreciate it.

**Comment by Rdgeno** [comment]
I dont know if this is going to make sense to you it doesn't to me. I was comparing rockyou to the other files the only diffrence was how far into the list the password was. I took a shot I changed it in one file then another and another everytime the password is in the first few thousand everylist gets it if I move it to far in it doesnt get it. Still no good with brute but straight list will get it as long as it's at the top of the list.

**Comment by Chick3nman** [comment]
First, to make it clear, we do not support the use of the GUI as it is a third party tool and not an official hashcat utility.

Second, I need you to produce a complete hashcat command and output from your environment. Is hashcat giving you errors/warnings? It still sounds like a runtime issue like I had originally guessed, however, I have almost nothing to go on without the complete command and output from hashcat.

**Comment by roycewilliams** [comment]
Third, it is much preferable for this kind of back-and-forth general fact-finding to happen in the forums, and then only once the issue is confirmed to be a bug, that an issue be opened on GitHub. Hence my original suggestion.

**Comment by Rdgeno** [comment]
First off sorry Mr. Williams I thought it would expedite the situation to just skip the attitudes and non answers and just go straight to the guys that know what they are doing. Yesterday was a bad day I finally finish my build first one so I went cheap first shot bad mobo then wrong sata you know how it goes starting out, and I fire it up and figure out something is wrong, then worse than it being the computer it might be hashcat and who knows for how long it's been going on. The months wasted I wasn't a happy camper yesterday, so if I was short I do apologize.
The commands are
-m 2500 /root/John.hccapx -a 3 2W43?d?d?d?d?d?d?d?d  -o/root/Documentsgotcha.txt the straight is -m 2500 -a 0 /root/John.hccapx -o /root/Documents/gotcha.txt /root/rockyou or any other list. The out put is either cracked or exhausted. It all depends where it is in the list. I'm not at home to look at it but I think the command is --opencl-device then the device number I have tried that and shut off my CPU same result but slower I have tried it with the GPU only the same result but fast. I apologize in advance if any of the commands are incorrect here I'm using my cellphone and cant see anything other than four or five words previous. Thank you again for taking notice of this problem I hope it's me not hashcat but as I have previously said I tried two computers. If there is an opencl or any other thing you would like me to try so I might help you just let me know and I'll get to it asap. Dealing with the frustration of cracking actually keeps me sane and I dont want hashcat to be broken. Have a good day and remember whatever you need just let me know and I'll try it.

**Comment by jsteube** [comment]
I don't see this going anywhere. Please try to describe your problem IN A CLEAN way in the hashcat forum, not the GitHub issue. GitHub issues are really just for bug reports and feature requests. If the guys in the hashcat forum approve that this is really a bug, then it's fine to open an issue here. Please try to understand otherwise developers are getting overwhelmed.

**Comment by Rdgeno** [comment]
I did like you said thank you, I was really thinking this was a developer issue. Sorry for wasting your time.
